bug: FFMPEG avcodec_close deprecated

**Describe the bug**
```
[10/203] Building CXX object replaygain/CMakeFiles/cantata-replaygain.dir/ffmpeginput.cpp.o
/home/nullobsi/Developer/cantata/replaygain/ffmpeginput.cpp:252:3: warning: 'avcodec_close' is deprecated [-Wdeprecated-declarations]
  252 |                 avcodec_close(handle->codecContext);
      |                 ^
/usr/include/libavcodec/avcodec.h:2386:1: note: 'avcodec_close' has been explicitly marked deprecated here
 2386 | attribute_deprecated
      | ^
/usr/include/libavutil/attributes.h:100:49: note: expanded from macro 'attribute_deprecated'
  100 | #    define attribute_deprecated __attribute__((deprecated))
      |                                                 ^
1 warning generated.
```

**Additional context**
Newer versions of ffmpeg have a different semantic about this. Need to check if the replacement function is implemented in older versions too.

--- a/gui/initialsettingswizard.cpp
+++ b/gui/initialsettingswizard.cpp
@@ -120,8 +120,11 @@ InitialSettingsWizard::InitialSettingsWizard(QWidget* p)
 	// Unix systems.
 #if !defined(Q_OS_WIN) && !defined(Q_OS_MACOS)
 	auto oldConfig = QDir(QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation) + "/cantata");
+	// We shouldn't be here if newConfig exists, but since there could
+	// be data-loss...
+	auto newConfig = QDir(QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation) + "/Cantata");
 
-	if (oldConfig.exists()) {
+	if (oldConfig.exists() && !newConfig.exists()) {
 		migrateDataBox->setChecked(true);
 	}
 	else {

--- a/replaygain/ffmpeginput.cpp
+++ b/replaygain/ffmpeginput.cpp
@@ -249,7 +249,7 @@ FfmpegInput::~FfmpegInput()
 {
 	if (handle) {
 		mutex.lock();
-		avcodec_close(handle->codecContext);
+		avcodec_free_context(&handle->codecContext);
 #if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(53, 21, 0)
 		avformat_close_input(&handle->formatContext);
 #else

